src/OSPSuite.Core/Services/PredictedVsObservedChartService.cs

```diff
@@ -107,6 +114,77 @@ public void ConfigureAxesDimensionAndTitle(IReadOnlyList<DataColumn> observation
          chart.UpdateAxesVisibility();
       }
 
+      public IEnumerable<DataRepository> AddDeviationLine(float foldValue, List<DataColumn> observationColumns, PredictedVsObservedChart chart,
+         int numberOfDeviationLines)
+      {
+         var deviationCurves = addDeviationLines(foldValue, observationColumns, chart, numberOfDeviationLines).ToList();
+         chart.UpdateAxesVisibility();
+         return deviationCurves;
+      }
+
+      private IEnumerable<DataRepository> addDeviationLines(float foldValue, IReadOnlyList<DataColumn> observationColumns,
+         PredictedVsObservedChart chart, int numberOfDeviationLines)
+      {
+         var dataColumns = observationColumns;
+
+         var uniqueDimensions = dataColumns.Select(dataColumn => _dimensionFactory.MergedDimensionFor(dataColumn))
+            .DistinctBy(dimension =>
+               dimension.DisplayName); //We are using display name here as it is the only way to identify unique merge dimensions
+
+         foreach (var mergedDimension in uniqueDimensions)
+         {
+            yield return createFoldDeviationRepositories(foldValue, chart, dataColumns, mergedDimension, numberOfDeviationLines);
+         }
+      }
+
+      private DataRepository createFoldDeviationRepositories(float foldValue, PredictedVsObservedChart chart,
+         IReadOnlyList<DataColumn> dataColumns,
+         IDimension mergedDimension, int numberOfDeviationLines)
+      {
+         var deviationLineRepository= createDeviationRepository(foldValue, dataColumns, mergedDimension);
+         chart.AddCurvesFor(deviationLineRepository, x => x.Name, _dimensionFactory,
+            (column, curve) => curve.UpdateDeviationCurve(column.Name, numberOfDeviationLines));
+
+         return deviationLineRepository;
+      }
+
+      private DataRepository createDeviationRepository(float foldValue, IReadOnlyList<DataColumn> dataColumns,
+         IDimension mergedDimension)
+      {
+         var columnsWithMatchingDimension = dataColumns.Where(x => x.Dimension.IsEquivalentTo(mergedDimension)).ToList();
+
+         var columnsForIdentityRepository = columnsWithMatchingDimension.Where(column => columnNonZeroValues(column).Any()).ToList();
+         if (!columnsForIdentityRepository.Any())
+            return null;
+
+         var identityMinimum = getIdentityMinimum(columnsForIdentityRepository, mergedDimension);
+         var identityMaximum = getIdentityMaximum(columnsForIdentityRepository, mergedDimension);
+
+         var baseGridUpper = new BaseGrid(Captions.ParameterIdentification.Deviation, mergedDimension)
```

Comment:
this should be rename. IT's called baseGridUpper but it's use for all lines

Reply:
done